src/Eshopworld.Core/Events/IEvent.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+namespace Eshopworld.Core.Events
+{
+    /// <summary>
+    /// Standard interface that every event that will be published & consumed from the event streaming bus will expose
+    /// </summary>
+    public interface IEvent
```

Comment:
Why is this an event as opposed to an abstract base class?

Reply:
An interface is used to standardize some of the base properties of each events, and is being used to narrow the scope of the event being emitted on the bus (including getting its identifier to be used as the message key). Do you want an additional base class to share / enforce certain properties as well? I've done this before but it's giving issues currently with the schema registry serialization & validation

Reply:
yeah I think that would be best as it gives us a place for future additions without breaking changes to class definition.

Reply:
Changed to baseevent now